docs/updater/updater.md

```diff
@@ -25,3 +25,7 @@ For the above listed reasons, we cannot simply use TUF's updater like described
 - New metadata files (not counting delegated roles) are acquired by calling TUF's `refresh`. Metadata files belonging to the delegated roles are updated by `get_current_targets`. Information about a target file is obtained using `get_one_valid_targetinfo`
 
 - Since TUF does not provide a way to gradually update metadata and target files (except for `root.json`), that has to be handled by TAF. So, TAF's updater calls TUF's updater multiple times. Our handler, `GitUpdateHandler` finds all commits that were pushed after the last time the client pulled the changes. The basic idea is to traverse trough these commits and update metadata files and targets for each of them, thus validating all revisions.
+
+- Validation and syncing before running the updater is crucial for the updater to work. Using `git pull`, we can ensure that both the authentication repository and target repositories are aligned with their respective last validated states. The process involves comparing the current state of the repositories on the client's machine with the latest state on the remote server and making necessary adjustments to bring them into sync. Examples of when validation and syncing is necessary when the client's authentication repository is not in sync with the remote repository, when target repositories are out of sync, when there are invalid or unauthorized commits, when repositories are in mixed states, and after manual updates. 
```

Comment:
I don't think that we should ever manually run gut pull before running th updater. The issue being addressed in this PR is a lot trickier, so instead of focusing on it, let's document what we did as a part of previous PRs (removal of additional commits, uncommitted changes etc.)